google-cloud-functions/src/main/java/com/google/cloud/functions/v1/nativeimage/CloudFunctionsFeature.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.functions.v1.nativeimage;
+
+import static com.google.api.gax.nativeimage.NativeImageUtils.registerClassForReflection;
+import static com.google.api.gax.nativeimage.NativeImageUtils.registerClassHierarchyForReflection;
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+import com.oracle.svm.core.configure.ResourcesRegistry;
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.hosted.Feature;
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+import org.graalvm.nativeimage.impl.ConfigurationCondition;
+
+/** A feature which registers reflective usages of the Cloud Functions library. */
+@AutomaticFeature
+final class CloudFunctionsFeature implements Feature {
+
+  private static final String FUNCTION_INVOKER_CLASS =
+      "com.google.cloud.functions.invoker.runner.Invoker";
```

Comment:
I think this class is part of https://github.com/GoogleCloudPlatform/functions-framework-java, which we decided not to pursue.

Reply:
Good point. All the configurations seem to be executed based on this class. I'm going to double check to see if they any of them relate to `java-functions` or if they are actually related.